src/Analyzer.JsonRuleEngine.UnitTests/AllOfExpressionDefinitionTests.cs

```diff
@@ -91,6 +129,42 @@ public void ToExpression_NestedAllOf_ReturnsArrayOfExpectedExpressions()
             Assert.IsInstanceOfType(allOfExpression.AllOf.Last(), typeof(LeafExpression));
         }
 
+        [TestMethod]
+        public void ToExpressionNegated_NestedAllOf_ReturnsArrayOfExpectedExpressions()
```

Comment:
Is it possible to collapse this test with the non-negated version? Same for the non-nested test and for the AnyOf tests

Reply:
Yeah it is possible and that's what I started out doing initially here, but I ultimately just split the tests out. Trying to have both scenarios run in one test would have been about the same number of lines as the two tests themselves, but a whole lot more convoluted. Since the negation causes the resultant expression object types to be different (AllOf/AnyOf), the rest of the code to verify the scenario gets less clean. Would involve lots of casting, using dynamic objects, or lots of if/else statements, etc.